com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeReferenceVolume.cs

```diff
@@ -691,36 +660,65 @@ public int GetVideoMemoryCost()
 
         void RemoveCell(Cell cell)
         {
-            if (cell.loaded)
+            if (cells.TryGetValue(cell.index, out var cellInfo))
             {
-                if (cells.ContainsKey(cell.index))
-                    cells.Remove(cell.index);
+                cells.Remove(cell.index);
 
-                if (m_ChunkInfo.ContainsKey(cell.index))
-                    m_ChunkInfo.Remove(cell.index);
+                if (cellInfo.loaded)
+                    m_LoadedCells.Remove(cellInfo);
+                else
+                    m_UnloadedCells.Remove(cellInfo);
 
-                if (cell.flatIdxInCellIndices >= 0)
-                    m_CellIndices.MarkCellAsUnloaded(cell.flatIdxInCellIndices);
+                UnloadCell(cellInfo);
 
-                RegId cellBricksID = new RegId();
-                if (m_CellToBricks.TryGetValue(cell, out cellBricksID))
-                {
-                    ReleaseBricks(cellBricksID);
-                    m_CellToBricks.Remove(cell);
-                }
+                m_CellInfoPool.Release(cellInfo);
             }
+        }
+
+        void UnloadCell(CellInfo cellInfo)
+        {
+            // Streaming might have never loaded the cell in the first place
+            if (cellInfo.loaded)
+            {
+                if (cellInfo.flatIdxInCellIndices >= 0)
+                    m_CellIndices.MarkCellAsUnloaded(cellInfo.flatIdxInCellIndices);
+
+                ReleaseBricks(cellInfo);
 
-            cell.loaded = false;
+                cellInfo.loaded = false;
+                cellInfo.updateInfo = new ProbeBrickIndex.CellIndexUpdateInfo();
+
+                ClearDebugData();
+            }
         }
 
-        void AddCell(Cell cell, List<Chunk> chunks)
+        internal void AddCell(Cell cell, int assetInstanceID)
         {
-            cell.loaded = true;
-            cells[cell.index] = cell;
+            // The same cell can exist in more than one asset
+            // Need to check existence because we don't want to add cells more than once to streaming structures
+            // TODO: Check perf if relevant?
```

Comment:
Is it? :D

Reply:
Kind of the same as above. This is relevant in term of perf when we have asset streaming. Right now it is expected that this happens during loading only so it's probably fine as is.